src/v/kafka/server/server.cc

```diff
@@ -236,10 +238,26 @@ ss::scheduling_group server::produce_scheduling_group() const {
              : ss::default_scheduling_group();
 }
 
+ss::scheduling_group server::get_request_handler_sg() const {
+    return config::shard_local_cfg().use_kafka_handler_scheduler_group()
+             ? _request_handler_scheduling_group
+             : ss::default_scheduling_group();
+}
+
 coordinator_ntp_mapper& server::coordinator_mapper() {
     return _group_router.local().coordinator_mapper().local();
 }
 
+ss::scheduling_group server::get_request_scheduling_group(api_key key) const {
+    if (key == produce_request::api_type::key) {
```

Comment:
For this kind of stuff usually we want to add a handler method I think, rather than having the kafka server know anything about the different keys. E.g., a get_request_sg() method on the generic handler which returns the group (or none/default) to use? 

Is the problem that the handlers shouldn't access config?